pkg/operator/controller/ingress/status.go

```diff
@@ -641,43 +691,42 @@ func formatConditions(conditions []*operatorv1.OperatorCondition) string {
-	condition := operatorv1.OperatorCondition{
-		Type:   operatorv1.OperatorStatusTypeProgressing,
-		Status: operatorv1.ConditionFalse,
+// computeIngressProgressingCondition computes the IngressController's current Progressing status state
+// by inspecting the following:
+// 1) the Deployment Replicas Updated condition of the IngressController
+// 2) the LoadBalancer Progressing condition of the IngressController
+// The IngressController is judged NOT Progressing only if all 2 conditions are true; otherwise
+// it is considered to be Progressing.
+func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition) operatorv1.OperatorCondition {
+	expected := []expectedCondition{
+		{
+			condition: IngressControllerLoadBalancerProgressingConditionType,
+			status:    operatorv1.ConditionFalse,
+		},
+		{
+			condition: IngressControllerDeploymentRollingOutConditionType,
+			status:    operatorv1.ConditionFalse,
+		},
 	}
-	if ic.Status.EndpointPublishingStrategy.Type == operatorv1.LoadBalancerServiceStrategyType {
-		switch {
-		case ic.Status.EndpointPublishingStrategy.LoadBalancer == nil:
-			condition.Message = "status.endpointPublishingStrategy.loadBalancer is not set."
-			condition.Reason = "IncompleteStatus"
-			condition.Status = operatorv1.ConditionUnknown
-		case service == nil:
-			condition.Message = "LoadBalancer Service not created."
-			condition.Reason = "NoService"
-			condition.Status = operatorv1.ConditionTrue
-		default:
-			wantScope := ic.Status.EndpointPublishingStrategy.LoadBalancer.Scope
-			haveScope := operatorv1.ExternalLoadBalancer
-			if IsServiceInternal(service) {
-				haveScope = operatorv1.InternalLoadBalancer
-			}
-			if wantScope != haveScope {
-				message := fmt.Sprintf("The IngressController scope was changed from %q to %q.", haveScope, wantScope)
-				switch platform.Type {
-				case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
-					message = fmt.Sprintf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created.  This will most likely cause the new load-balancer to have a different host name and IP address from the old one's.  Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/%[4]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"loadBalancer\":{\"scope\":\"%[5]s\"}}}}'", message, service.Namespace, service.Name, ic.Name, haveScope)
-				}
-				condition.Reason = "ScopeChanged"
-				condition.Message = message
-				condition.Status = operatorv1.ConditionTrue
-			}
+
+	// Cover the rare case of no conditions
+	if len(conditions) == 0 {
+		return operatorv1.OperatorCondition{Type: operatorv1.OperatorStatusTypeProgressing, Status: operatorv1.ConditionTrue}
```

Comment:
I'm 50/50 on whether this should be ConditionTrue or ConditionFalse if all of the conditions don't exist

But if 1 condition exists, and the other is missing, it doesn't account for the the missing condition. Seems a bit arbitrary.

Reply:
The default from the previous incarnation of this method was to return ConditionFalse; that probably shouldn't change.

Reply:
Does that mean you're going to change it?

Reply:
Whoops sorry, I misinterpreted. I think it's slightly an apples-to-oranges comparison. The previous method did return false by default, but it was at least was attempting to calculate the progressing status.

On the other hand, this logic says "if there are no `IngressControllerLoadBalancerProgressingConditionType` or `IngressControllerDeploymentRollingOutConditionType` status conditions, then say we are progressing". It determines progressing based on if the other conditions exist (which, by all means, should always exist, unless something is wrong).

Like I said, it seems a bit arbitrary, but I suppose we should err on the side of not progressing if the conditions don't exist, so I will change it.